src/coreclr/jit/assertionprop.cpp

```diff
@@ -3389,6 +3389,17 @@ GenTree* Compiler::optConstantAssertionProp(AssertionDsc*        curAssertion,
             }
             break;
 
+        case O2K_ZEROOBJ:
```

Comment:
I don't quite understand this addition. Not all consumers of struct values support "zero" values, it's the reason for the odd `ZEROOBJ` setup with propagating on parents, after all.

Reply:
Yeah, not needed.